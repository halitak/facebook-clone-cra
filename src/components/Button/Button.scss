.Button {
  display: flex;
  height: 40px;
  align-items: center;
  padding-right: 8px;
  padding-left: 8px;
  background-color: transparent;
  border-radius: 8px;
  color: inherit;
  cursor: pointer;
  transition: var(--fade);
  &:hover {
    background-color: var(--c-hover);
  }
  > span {
    margin-left: 8px;
  }

  &-small {
    height: 36px;
    padding: 4px;
    > span {
      margin-right: 8px;
      margin-left: 6px;
    }
  }
  &-big{
    height: 44px;
    > span {
      margin-left: 10px;
    }
  }
  &-xbig{
    height: 48px;
    > span {
      margin-left: 10px;
    }
  }

  &-figure {
    height: auto;
    justify-content: center;
    padding: 0;
    border-radius: 50%;
  }
}
