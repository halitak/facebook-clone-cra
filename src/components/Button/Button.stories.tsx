import React from 'react';
import { storiesOf } from '@storybook/react';

import Button from './Button';
import Figure from '../Figure';
import { Close } from '../Icons';
import { PROFILE_PHOTO } from '../../constants';

storiesOf('Button', module)
  .add('Normal', () => <Button>Normal Button</Button>)
  .add('Small', () => <Button small>Small Button</Button>)
  .add('Figure', () => (
    <div className="h-stack">
      <Button figure>
        <Figure small>
          <img src={PROFILE_PHOTO.src} alt={PROFILE_PHOTO.title} />
        </Figure>
      </Button>
      <Button figure>
        <Figure>
          <img src={PROFILE_PHOTO.src} alt={PROFILE_PHOTO.title} />
        </Figure>
      </Button>
      <Button figure>
        <Figure big>
          <img src={PROFILE_PHOTO.src} alt={PROFILE_PHOTO.title} />
        </Figure>
      </Button>
      <Button figure>
        <Figure small>
          <Close />
        </Figure>
      </Button>
      <Button figure>
        <Figure>
          <Close />
        </Figure>
      </Button>
      <Button figure>
        <Figure big>
          <Close />
        </Figure>
      </Button>
    </div>
  ));
