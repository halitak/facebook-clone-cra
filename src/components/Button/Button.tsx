import React, { FC } from 'react';
import cx from 'classnames';

import './Button.scss';

type Props = JSX.IntrinsicElements['button'] & {
  small?: boolean;
  big?: boolean;
  xbig?: boolean;
  figure?: boolean;
};

const Button: FC<Props> = ({ small, big, xbig, figure, className, children, onClick, ...props }) => {
  const classNames = cx(
    'Button',
    {
      'Button-small': small,
      'Button-big': big,
      'Button-xbig': xbig,
      'Button-figure': figure,
    },
    className,
  );

  return (
    <button className={classNames} onClick={onClick} {...props}>
      {children}
    </button>
  );
};

export default Button;
