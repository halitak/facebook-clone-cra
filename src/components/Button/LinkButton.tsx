import React, { FC } from 'react';
import cx from 'classnames';
import { Link } from 'react-router-dom';

import './Button.scss';

type Props = JSX.IntrinsicElements['a'] & {
  xsmall?: boolean;
  small?: boolean;
  big?: boolean;
  xbig?: boolean;
  figure?: boolean;
  href?: any;
};

const LinkButton: FC<Props> = ({ xsmall, small, big, xbig, figure, href, className, children }) => {
  const link = href ? true : false;
  const classNames = cx(
    'Button',
    {
      "Button-link": link,
      'Button-xsmall': xsmall,
      'Button-small': small,
      'Button-big': big,
      'Button-xbig': xbig,
      'Button-figure': figure,
    },
    className,
  );

  return (
    <Link to={href} className={classNames}>
      {children}
    </Link>
  );
};

export default LinkButton;
