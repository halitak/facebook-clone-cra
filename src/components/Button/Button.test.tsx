import React from 'react';
import { shallow } from 'enzyme';

import Button from './Button';

it('renders correctly normal button', () => {
  const wrapper = shallow(<Button>Normal Button</Button>);
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly small button', () => {
  const wrapper = shallow(<Button small>Small Button</Button>);
  expect(wrapper).toMatchSnapshot();
});

it('renders correctly small button', () => {
  const wrapper = shallow(<Button figure>Figure Button</Button>);
  expect(wrapper).toMatchSnapshot();
});
