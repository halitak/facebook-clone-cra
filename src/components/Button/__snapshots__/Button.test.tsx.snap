// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly normal button 1`] = `
<button
  className="Button"
>
  Normal Button
</button>
`;

exports[`renders correctly small button 1`] = `
<button
  className="Button Button-small"
>
  Small Button
</button>
`;

exports[`renders correctly small button 2`] = `
<button
  className="Button Button-figure"
>
  Figure Button
</button>
`;
